Tests/SwiftDocCTests/Infrastructure/SymbolGraph/SymbolGraphLoaderTests.swift

```diff
@@ -1506,6 +1506,218 @@ class SymbolGraphLoaderTests: XCTestCase {
+                ],
+                "names": {
+                    "title": "Foo",
+                },
+                "accessLevel": "public",
+                "availability" : [
+                    {
+                      "domain" : "iOS",
+                      "introduced" : {
+                        "major" : 12,
+                        "minor" : 0,
+                        "patch" : 0
+                      }
+                    }
+                ]
+            }
+            """,
+            platform: """
+            "environment" : "macabi",
+            "operatingSystem" : {
+               "minimumVersion" : {
+                 "major" : 6,
+                 "minor" : 5,
+                 "patch" : 0
+               },
+               "name" : "ios"
+             }
+            """
+        )
+        let infoPlist = """
+        <plist version="1.0">
+        <dict>
+            <key>CFBundleDisplayName</key>
+            <string>MyModule</string>
+            <key>CFBundleIdentifier</key>
+            <string>com.apple.MyModule</string>
+            <key>CFBundleVersion</key>
+            <string>0.1.0</string>
+            <key>CDAppleDefaultAvailability</key>
+            <dict>
+                <key>MyModule</key>
+                <array>
+                    <dict>
+                        <key>name</key>
+                        <string>Mac Catalyst</string>
+                        <key>version</key>
+                        <string>1.0</string>
+                    </dict>
+                </array>
+            </dict>
+        </dict>
+        </plist>
+        """
+        // Create an empty bundle
+        let targetURL = try createTemporaryDirectory(named: "test.docc")
+        // Store files
+        try symbolGraphStringiOS.write(to: targetURL.appendingPathComponent("MyModule-ios.symbols.json"), atomically: true, encoding: .utf8)
+        try symbolGraphStringCatalyst.write(to: targetURL.appendingPathComponent("MyModule-catalyst.symbols.json"), atomically: true, encoding: .utf8)
+        try infoPlist.write(to: targetURL.appendingPathComponent("Info.plist"), atomically: true, encoding: .utf8)
+        // Load the bundle & reference resolve symbol graph docs
+        let (_, _, context) = try loadBundle(from: targetURL)
+        guard let availability = (context.documentationCache["c:@F@A"]?.semantic as? Symbol)?.availability?.availability else {
+            XCTFail("Did not find availability for symbol 'c:@F@A'")
+            return
+        }
+        // Verify we fallback to iOS even if there's default availability for the Catalyst platform.
+        XCTAssertNotNil(availability.first(where: { $0.domain?.rawValue == "iOS" }))
+        XCTAssertNotNil(availability.first(where: { $0.domain?.rawValue == "macCatalyst" }))
+        XCTAssertEqual(availability.first(where: { $0.domain?.rawValue == "macCatalyst" })?.introducedVersion, SymbolGraph.SemanticVersion(major: 12, minor: 0, patch: 0))
```

Comment:
minor: Since the same value is used in the assert-equal check below, you could either unwrap it first and compare it second:
```suggestion
        let catalystAvailability = try XCTUnwrap(availability.first(where: { $0.domain?.rawValue == "macCatalyst" }))
        XCTAssertEqual(catalystAvailability.introducedVersion, SymbolGraph.SemanticVersion(major: 12, minor: 0, patch: 0))
```

or you could skip the assert-not-nil check. If the availability value is `nil`, its introduced version will be `nil` through optional chaining which won't be equal to "12.0.0"

```suggestion
        XCTAssertEqual(availability.first(where: { $0.domain?.rawValue == "macCatalyst" })?.introducedVersion, SymbolGraph.SemanticVersion(major: 12, minor: 0, patch: 0))
```